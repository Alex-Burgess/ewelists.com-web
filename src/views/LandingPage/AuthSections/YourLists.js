import React, { useState, useEffect } from 'react';
// libs
import { onError } from "libs/errorLib";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
import Icon from "@material-ui/core/Icon";
// @material-ui/icons
import ViewIcon from "@material-ui/icons/Visibility";
// core components
import GridContainer from "components/Grid/GridContainer.js";
import GridItem from "components/Grid/GridItem.js";
import Button from "components/CustomButtons/Button.js";
import Card from "components/Card/Card.js";
import CardHeader from "components/Card/CardHeader.js";
import CardBody from "components/Card/CardBody.js";
import CardFooter from "components/Card/CardFooter.js";
import Info from "components/Typography/Info.js";
// Custom Buttons
import CreateButton from "custom/Buttons/CreateButton.js";
import CreatePopOut from "./CreatePopOut.js";

import { GetLists } from "./YourListsApi";

import styles from "assets/jss/custom/views/landingPage/yourListsStyle.js";
const useStyles = makeStyles(styles);

export default function YourLists(props) {
  const classes = useStyles();

  const { showCreate } = props;

  const [isLoading, setIsLoading] = useState(true);
  const [openLists, setOpenLists] = useState([]);
  const [closedLists, setClosedLists] = useState([]);

  useEffect( () => {
    async function getLists(){
      let response;
      try {
        response = await GetLists();
        setOpenLists(response.owned);
        setClosedLists(response.closed);
      } catch (e) {
        onError(e);
      }
    }

    getLists();
    setIsLoading(false);
  }, []);

  const renderOpenLists = (lists: Lists[]) => {
    let allLists: Lists[] = [];

    return allLists.concat(lists).map (
      (list, i) =>
        <GridItem xs={12} sm={4} md={4} key={i}>
          <Card profile>
            <CardHeader image>
              <a href={"/lists/" + list.listId}>
                <img src={list.imageUrl} className={classes.listImage} alt="..." />
              </a>
              <div
                className={classes.coloredShadow}
                style={{
                  backgroundImage: `url(${list.imageUrl})`,
                  opacity: "1"
                }}
              />
            </CardHeader>
            <CardBody>
              <Info>
                <a href={"/lists/" + list.listId}>
                  <h6 className={classes.cardCategory}>{list.title}</h6>
                </a>
              </Info>
              <p className={classes.cardDescription}>
                {list.occasion}
              </p>
            </CardBody>
              <CardFooter profile className={classes.justifyContentCenter}>
                <a href={"/lists/" + list.listId}>
                  <Button round color="info" className={classes.customButton}>
                    <ViewIcon /> View
                  </Button>
                </a>
                <a href={"/edit/" + list.listId}>
                  <Button round color="success" className={classes.customButton}>
                    <Icon>mode_edit</Icon> Edit
                  </Button>
                </a>
              </CardFooter>
          </Card>
        </GridItem>
    )
  }

  const renderClosedLists = (lists: Lists[]) => {
    let allLists: Lists[] = [];

    return allLists.concat(lists).map (
      (list, i) =>
        <GridItem xs={12} sm={4} md={4} key={i}>
          <Card profile>
            <CardHeader image>
              <a href={"/lists/" + list.listId}>
                <img src={list.imageUrl} className={classes.listImage} alt="..." />
              </a>
              <div
                className={classes.coloredShadow}
                style={{
                  backgroundImage: `url(${list.imageUrl})`,
                  opacity: "1"
                }}
              />
            </CardHeader>
            <CardBody>
              <Info>
                <a href={"/lists/" + list.listId}>
                  <h6 className={classes.cardCategory}>{list.title}</h6>
                </a>
              </Info>
              <p className={classes.cardDescription}>
                {list.occasion} - closed
              </p>
            </CardBody>
              <CardFooter profile className={classes.justifyContentCenter}>
                <a href={"/edit/" + list.listId}>
                  <Button round color="info" className={classes.customButton}>
                    <Icon>mode_edit</Icon> Details
                  </Button>
                </a>
              </CardFooter>
          </Card>
        </GridItem>
    )
  }

  return (
    <div className={classes.section}>
      <div className={classes.container}>
        <h1 className={classes.title}>Your lists</h1>
        <br />
        <GridContainer>
          {!isLoading && renderOpenLists(openLists)}
          {!isLoading && renderClosedLists(closedLists)}
          <GridItem xs={12} sm={4} md={4}>
            <CreateButton
              text="Create a New List"
              onClick={() => props.setCreate(true)}
            />
          {<CreatePopOut open={showCreate} setCreate={props.setCreate} />}
          </GridItem>
        </GridContainer>
      </div>
    </div>
  );
}

YourLists.propTypes = {
  showCreate: PropTypes.bool,
};
