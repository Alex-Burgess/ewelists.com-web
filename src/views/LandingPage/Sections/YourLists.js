import React from "react";
import { API } from "aws-amplify";
import { withRouter } from 'react-router-dom'
// nodejs library to set properties for components
import PropTypes from "prop-types";
// @material-ui/core components
import withStyles from "@material-ui/core/styles/withStyles";
import Icon from "@material-ui/core/Icon";
import Slide from "@material-ui/core/Slide";
import Dialog from "@material-ui/core/Dialog";
import DialogTitle from "@material-ui/core/DialogTitle";
import DialogContent from "@material-ui/core/DialogContent";
import InputLabel from "@material-ui/core/InputLabel";
import MenuItem from "@material-ui/core/MenuItem";
import FormControl from "@material-ui/core/FormControl";
import Select from "@material-ui/core/Select";
// @material-ui/icons
import Subject from "@material-ui/icons/Subject";
import Close from "@material-ui/icons/Close";
// core components
import GridContainer from "components/Grid/GridContainer.js";
import GridItem from "components/Grid/GridItem.js";
import Button from "components/CustomButtons/Button.js";
import Card from "components/Card/Card.js";
import CardHeader from "components/Card/CardHeader.js";
import CardBody from "components/Card/CardBody.js";
import CardFooter from "components/Card/CardFooter.js";
import Info from "components/Typography/Info.js";
import CustomInput from "components/CustomInput/CustomInput.js";
import CircularProgress from '@material-ui/core/CircularProgress';
// Custom Buttons
import CreateButton from "custom/Buttons/CreateButton.js";


import createStyle from "assets/jss/custom/views/landingPage/yourListsStyle.js";
import config from 'config.js';

const Transition = React.forwardRef(function Transition(props, ref) {
  return <Slide direction="down" ref={ref} {...props} />;
});

Transition.displayName = "Transition";

class SectionLists extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      isCreating: false,
      isLoading: true,
      ownedLists: [],
      shareLists: [],
      createModal: false,
      title: '',
      description: '',
      occasion: '',
      occasionList: ['Baby Shower', 'Birthday', 'Christmas', 'Baptism', 'Christening', 'Other']
    };
  }

  async componentDidMount() {
    const response = await this.getLists();
    this.setState({ ownedLists: response.owned });
    this.setState({ sharedLists: response.shared });
    this.setState({ isLoading: false });
  }

  getLists() {
    return API.get("lists", "/");
  }

  handleClickOpen(modal) {
    var x = [];
    x[modal] = true;
    this.setState(x);
  }
  handleClose(modal) {
    var x = [];
    x[modal] = false;
    this.setState(x);
  }

  handleSimple = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  changeHandler = event => {
    this.setState({
      [event.target.id]: event.target.value
    });
  }

  validateForm() {
    return (
      this.state.title.length > 0 &&
      this.state.description.length > 0 &&
      this.state.occasion.length > 0
    );
  }

  createList = async event => {
    this.setState({ isCreating: true });

    const occasion_parsed = this.state.occasion.toLowerCase().replace(/\s/g,'');
    const new_image_url = config.imagePrefix + '/images/' + occasion_parsed + '-default.jpg';

    console.log("Creating list with values: title: " + this.state.title + ", description: " + this.state.description + ", occasion: " + this.state.occasion + ", imageUrl: " + new_image_url);

    var createList = {
      "title": this.state.title,
      "description": this.state.description,
      "occasion": this.state.occasion,
      "imageUrl": new_image_url
    }

    const response = await this.createListRequest(createList);

    const listId = response.listId;
    console.log("List was created with ID (" + listId + "), redirecting to edit page for list.")

    this.props.history.push('/edit/' + listId);
  }

  createListRequest(createList) {
    return API.post("lists", "/", {
      body: createList
    });
  }

  renderYourLists(classes, lists: Lists[], shared) {
    let allLists: Lists[] = [];

    return allLists.concat(lists).map (
      (list, i) =>
        <GridItem xs={12} sm={4} md={4} key={i}>
          <Card profile>
            <CardHeader image>
              <a href={"/lists/" + list.listId}>
                <img src={list.imageUrl} className={classes.listImage} alt="..." />
              </a>
              <div
                className={classes.coloredShadow}
                style={{
                  backgroundImage: `url(${list.imageUrl})`,
                  opacity: "1"
                }}
              />
            </CardHeader>
            <CardBody>
              <Info>
                <a href={"/lists/" + list.listId}>
                  <h6 className={classes.cardCategory}>{list.title}</h6>
                </a>
              </Info>
              <p className={classes.cardDescription}>
                {list.occasion}
              </p>
            </CardBody>
            {shared
              ? <CardFooter profile className={classes.justifyContentCenter}>
                  <a href={"/lists/" + list.listId}>
                    <Button round color="info" className={classes.reserveButton2}>
                      Shop List
                    </Button>
                  </a>
                </CardFooter>
              : <CardFooter profile className={classes.justifyContentCenter}>
                  <a href={"/lists/" + list.listId}>
                    <Button round justIcon color="info">
                      <Subject />
                    </Button>
                  </a>
                  <a href={"/edit/" + list.listId}>
                    <Button round justIcon color="success">
                      <Icon>mode_edit</Icon>
                    </Button>
                  </a>
                </CardFooter>
            }
          </Card>
        </GridItem>
    )
  }

  renderOccasionSelect(classes) {
    return (
      <FormControl fullWidth className={classes.selectFormControl}>
        <Select
          MenuProps={{className: classes.selectMenu}}
          classes={{select: classes.select}}
          value={this.state.occasion}
          onChange={this.handleSimple}
          inputProps={{name: "occasion", id: "simple-select"}}
        >
          <MenuItem disabled classes={{root: classes.selectMenuItem}}>
            Select Type
          </MenuItem>
          {this.renderMenuItems(classes, this.state.occasionList)}
        </Select>
      </FormControl>
    )
  }

  renderMenuItems(classes, occasionList: Occasions[]) {
    return occasionList.map(
      (label, i) =>
          <MenuItem classes={{root: classes.selectMenuItem, selected: classes.selectMenuItemSelected}} value={label} key={i}>
            {label}
          </MenuItem>
    )
  }

  renderCreatePopOut(classes){
    return (
      <Dialog
        classes={{
          root: classes.modalRoot,
          paper: classes.modal + " " + classes.modalClassic
        }}
        open={this.state.createModal}
        TransitionComponent={Transition}
        keepMounted
        onClose={() => this.handleClose("createModal")}
        aria-labelledby="classic-modal-slide-title"
        aria-describedby="classic-modal-slide-description"
      >
        <DialogTitle
          id="classic-modal-slide-title"
          disableTypography
          className={classes.modalHeader}
        >
          <Button
            simple
            className={classes.modalCloseButton}
            key="close"
            aria-label="Close"
            onClick={() => this.handleClose("createModal")}
          >
            {" "}
            <Close className={classes.modalClose} />
          </Button>
          <h3
            className={
              classes.cardTitle + " " + classes.modalTitle + " " + classes.textCenter
            }
          >
            Create New List
          </h3>
        </DialogTitle>
        <DialogContent
          id="classic-modal-slide-description"
          className={classes.modalBody}
        >
          <GridContainer>
            <GridItem
              xs={12}
              sm={12}
              md={12}
              className={classes.mrAuto}
            >
              <form className={classes.form}>
                <InputLabel className={classes.label}>
                  Title:
                </InputLabel>
                <CustomInput
                  id="title"
                  inputProps={{
                    placeholder: "Enter your title here...",
                    onChange: this.changeHandler
                  }}
                  formControlProps={{
                    fullWidth: true
                  }}
                />
                <InputLabel className={classes.label}>
                  Description:
                </InputLabel>
                <CustomInput
                  id="description"
                  formControlProps={{
                    fullWidth: true
                  }}
                  inputProps={{
                    placeholder: "Add your description here...",
                    onChange: this.changeHandler
                  }}
                />
                <InputLabel className={classes.label + " " + classes.date}>
                  Occasion:
                </InputLabel>
                {this.renderOccasionSelect(classes)}
                <div className={classes.root}>
                  <div className={classes.wrapper}>
                    <Button
                      variant="contained"
                      color="primary"
                      className={classes.buttonSuccess}
                      disabled={!this.validateForm() || this.state.isCreating}
                      onClick={this.createList}
                    >
                      Create
                    </Button>
                    {this.state.isCreating && <CircularProgress size={24} className={classes.buttonProgress} />}
                  </div>
                </div>
              </form>
            </GridItem>
          </GridContainer>
        </DialogContent>
      </Dialog>
    )
  }


  render() {
    const { classes } = this.props;
    return (
      <div className={classes.section}>
        <div className={classes.container}>
          <GridContainer>
            <GridItem xs={12} sm={12} md={12}>
              <h1 className={classes.title}>Your lists</h1>
              <br />
              <GridContainer>
                {!this.state.isLoading && this.renderYourLists(classes, this.state.ownedLists, false)}
                <GridItem xs={12} sm={4} md={4}>
                  <CreateButton
                    text="Create a New List"
                    onClick={() => this.handleClickOpen("createModal")}
                  />
                  {this.renderCreatePopOut(classes)}
                </GridItem>
              </GridContainer>
            </GridItem>
            <GridItem xs={12} sm={12} md={12}>
              <h1 className={classes.title}>Lists Shared With You</h1>
              <br />
              <GridContainer>
                {!this.state.isLoading && this.renderYourLists(classes, this.state.sharedLists, true)}
              </GridContainer>
            </GridItem>
          </GridContainer>
        </div>
      </div>
    );
  }
}

SectionLists.propTypes = {
  classes: PropTypes.object
};

export default withRouter(withStyles(createStyle)(SectionLists));
