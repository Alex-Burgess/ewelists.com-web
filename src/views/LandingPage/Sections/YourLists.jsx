/*!

=========================================================
* Material Kit PRO React - v1.7.0
=========================================================

* Product Page: https://www.creative-tim.com/product/material-kit-pro-react
* Copyright 2019 Creative Tim (https://www.creative-tim.com)

* Coded by Creative Tim

=========================================================

* The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

*/
import React from "react";
import { API } from "aws-amplify";
import { withRouter } from 'react-router-dom'
// nodejs library to set properties for components
import PropTypes from "prop-types";
// @material-ui/core components
import withStyles from "@material-ui/core/styles/withStyles";
import Icon from "@material-ui/core/Icon";
import Slide from "@material-ui/core/Slide";
import Dialog from "@material-ui/core/Dialog";
import DialogTitle from "@material-ui/core/DialogTitle";
import DialogContent from "@material-ui/core/DialogContent";
import InputLabel from "@material-ui/core/InputLabel";
import MenuItem from "@material-ui/core/MenuItem";
import FormControl from "@material-ui/core/FormControl";
import Select from "@material-ui/core/Select";
// @material-ui/icons
import Subject from "@material-ui/icons/Subject";
import Close from "@material-ui/icons/Close";
// core components
import GridContainer from "components/Grid/GridContainer.jsx";
import GridItem from "components/Grid/GridItem.jsx";
import Button from "components/CustomButtons/Button.jsx";
import CreateButton from "components/CreateButton/CreateButton.jsx";
import Card from "components/Card/Card.jsx";
import CardHeader from "components/Card/CardHeader.jsx";
import CardBody from "components/Card/CardBody.jsx";
import CardFooter from "components/Card/CardFooter.jsx";
import Info from "components/Typography/Info.jsx";
import CustomInput from "components/CustomInput/CustomInput.jsx";
import CircularProgress from '@material-ui/core/CircularProgress';

import createStyle from "assets/jss/material-kit-pro-react/views/landingPageSections/yourListsStyle.jsx";

import oscar1 from "assets/img/examples/oscar-birthday.jpg";
// import oscar2 from "assets/img/examples/oscar-christmas.jpg";

const Transition = React.forwardRef(function Transition(props, ref) {
  return <Slide direction="down" ref={ref} {...props} />;
});

Transition.displayName = "Transition";

class SectionLists extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      isCreating: false,
      isLoading: true,
      lists: [],
      createModal: false,
      title: '',
      description: '',
      occasion: ''
    };
  }

  async componentDidMount() {
    const response = await this.getLists();
    this.setState({ lists: response.lists });
    this.setState({ isLoading: false });
  }

  getLists() {
    return API.get("lists", "/");
  }

  handleClickOpen(modal) {
    var x = [];
    x[modal] = true;
    this.setState(x);
  }
  handleClose(modal) {
    var x = [];
    x[modal] = false;
    this.setState(x);
  }

  handleSimple = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  changeHandler = event => {
    this.setState({
      [event.target.id]: event.target.value
    });
  }

  createList = async event => {
    this.setState({ isCreating: true });

    console.log("Creating list with values: " + this.state.title + " " + this.state.description + " " + this.state.occasion);

    var createList = {
      "title": this.state.title,
      "description": this.state.description,
      "occasion": this.state.occasion
    }

    const response = await this.createListRequest(createList);

    const listId = response.listId;
    console.log("List was created with ID (" + listId + "), redirecting to edit page for list.")

    this.props.history.push('/edit/' + listId);
  }

  createListRequest(createList) {
    return API.post("lists", "/", {
      body: createList
    });
  }

  renderYourLists(classes, lists: Lists[]) {
    let allLists: Lists[] = [];

    return allLists.concat(lists).map (
      (list, i) =>
        <GridItem xs={12} sm={4} md={4} key={i}>
          <Card profile>
            <CardHeader image>
              <a href="/lists/viewexample">
                <img src={oscar1} className={classes.listImage} alt="..." />
              </a>
              <div
                className={classes.coloredShadow}
                style={{
                  backgroundImage: `url(${oscar1})`,
                  opacity: "1"
                }}
              />
            </CardHeader>
            <CardBody>
              <Info>
                <a href="/lists/viewexample">
                  <h6 className={classes.cardCategory}>{list.title}</h6>
                </a>
              </Info>
              <p className={classes.cardDescription}>
                {list.description}
              </p>
            </CardBody>
            <CardFooter
              profile
              className={classes.justifyContentCenter}
            >
              <a href="/lists/viewexample">
                <Button round justIcon color="info">
                  <Subject />
                </Button>
              </a>
              <a href={"/edit/" + list.listId}>
                <Button round justIcon color="success">
                  <Icon>mode_edit</Icon>
                </Button>
              </a>
            </CardFooter>
          </Card>
        </GridItem>
    )
  }

  render() {
    const { classes } = this.props;
    return (
      <div className={classes.section}>
        <div className={classes.container}>
          <GridContainer>
            <GridItem xs={12} sm={12} md={12}>
              <h1 className={classes.title}>Your lists</h1>
              <br />
              <GridContainer>
                {!this.state.isLoading && this.renderYourLists(classes, this.state.lists)}
                <GridItem xs={12} sm={4} md={4}>
                  <CreateButton
                    text="Create a New List"
                    onClick={() => this.handleClickOpen("createModal")}
                  />
                  {/* CREATE MODAL START */}
                  <Dialog
                    classes={{
                      root: classes.modalRoot,
                      paper: classes.modal + " " + classes.modalClassic
                    }}
                    open={this.state.createModal}
                    TransitionComponent={Transition}
                    keepMounted
                    onClose={() => this.handleClose("createModal")}
                    aria-labelledby="classic-modal-slide-title"
                    aria-describedby="classic-modal-slide-description"
                  >
                    <DialogTitle
                      id="classic-modal-slide-title"
                      disableTypography
                      className={classes.modalHeader}
                    >
                      <Button
                        simple
                        className={classes.modalCloseButton}
                        key="close"
                        aria-label="Close"
                        onClick={() => this.handleClose("createModal")}
                      >
                        {" "}
                        <Close className={classes.modalClose} />
                      </Button>
                      <h3
                        className={
                          classes.cardTitle + " " + classes.modalTitle + " " + classes.textCenter
                        }
                      >
                        Create New List
                      </h3>
                    </DialogTitle>
                    <DialogContent
                      id="classic-modal-slide-description"
                      className={classes.modalBody}
                    >
                      <GridContainer>
                        <GridItem
                          xs={12}
                          sm={12}
                          md={12}
                          className={classes.mrAuto}
                        >
                          <form className={classes.form}>
                            <InputLabel className={classes.label}>
                              Title:
                            </InputLabel>
                            <CustomInput
                              id="title"
                              smallTitle
                              inputProps={{
                                placeholder: "Enter your title here...",
                                onChange: this.changeHandler
                              }}
                              formControlProps={{
                                fullWidth: true
                              }}
                            />
                            <InputLabel className={classes.label}>
                              Description:
                            </InputLabel>
                            <CustomInput
                              id="description"
                              smallDescription
                              formControlProps={{
                                fullWidth: true
                              }}
                              inputProps={{
                                placeholder: "Add your description here...",
                                onChange: this.changeHandler
                              }}
                            />
                            <InputLabel className={classes.label + " " + classes.date}>
                              Occasion:
                            </InputLabel>
                            <FormControl fullWidth className={classes.selectFormControl}>
                              <Select
                                MenuProps={{
                                  className: classes.selectMenu
                                }}
                                classes={{
                                  select: classes.select
                                }}
                                value={this.state.occasion}
                                onChange={this.handleSimple}
                                inputProps={{
                                  name: "occasion",
                                  id: "simple-select"
                                }}
                              >
                                <MenuItem
                                  disabled
                                  classes={{
                                    root: classes.selectMenuItem
                                  }}
                                >
                                  Select Type
                                </MenuItem>
                                <MenuItem
                                  classes={{
                                    root: classes.selectMenuItem,
                                    selected: classes.selectMenuItemSelected
                                  }}
                                  value="Baby Shower"
                                >
                                  Baby Shower
                                </MenuItem>
                                <MenuItem
                                  classes={{
                                    root: classes.selectMenuItem,
                                    selected: classes.selectMenuItemSelected
                                  }}
                                  value="Birthday"
                                >
                                  Birthday
                                </MenuItem>
                                <MenuItem
                                  classes={{
                                    root: classes.selectMenuItem,
                                    selected: classes.selectMenuItemSelected
                                  }}
                                  value="Christmas"
                                >
                                  Christmas
                                </MenuItem>
                                <MenuItem
                                  classes={{
                                    root: classes.selectMenuItem,
                                    selected: classes.selectMenuItemSelected
                                  }}
                                  value="Baptism"
                                >
                                  Baptism
                                </MenuItem>
                                <MenuItem
                                  classes={{
                                    root: classes.selectMenuItem,
                                    selected: classes.selectMenuItemSelected
                                  }}
                                  value="Christening"
                                >
                                  Christening
                                </MenuItem>
                                <MenuItem
                                  classes={{
                                    root: classes.selectMenuItem,
                                    selected: classes.selectMenuItemSelected
                                  }}
                                  value="Other"
                                >
                                  Other
                                </MenuItem>
                              </Select>
                            </FormControl>
                            <div className={classes.root}>
                              <div className={classes.wrapper}>
                                <Button
                                  variant="contained"
                                  color="primary"
                                  className={classes.buttonSuccess}
                                  disabled={this.state.isCreating}
                                  onClick={this.createList}
                                >
                                  Create
                                </Button>
                                {this.state.isCreating && <CircularProgress size={24} className={classes.buttonProgress} />}
                              </div>
                            </div>
                          </form>
                        </GridItem>
                      </GridContainer>
                    </DialogContent>
                  </Dialog>
                  {/* CREATE MODAL END */}
                </GridItem>
              </GridContainer>
            </GridItem>
          </GridContainer>
        </div>
      </div>
    );
  }
}

SectionLists.propTypes = {
  classes: PropTypes.object
};

export default withRouter(withStyles(createStyle)(SectionLists));
