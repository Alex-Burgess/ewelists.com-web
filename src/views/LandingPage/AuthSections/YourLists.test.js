import * as React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import {mount} from 'enzyme';
import YourLists from './YourLists';

describe('Tests for YourLists section.', () => {
  const api = require('./YourListsApi.js');
  api.GetLists = jest.fn(() => {
    return {
      "user": {
          "email": "burgess.alexander+apitestuser@gmail.com",
          "userId": "ae617d08-1127-4066-87ec-b6df345793b8",
          "name": "Api User"
      },
      "owned": [
          {
              "listId": "933c7476-cf0d-4a75-8822-2c011c07ce1d",
              "title": "My Birthday List",
              "description": "A gift wish list for my birthday.",
              "occasion": "Birthday",
              "imageUrl": "/images/celebration-default.jpg",
              "listOwner": "ae617d08-1127-4066-87ec-b6df345793b8"
          }
      ],
      "shared": []
    }
  });

  test('Renders your lists section', async () => {
    let component;

    const {act} = TestRenderer;

    await act( async () => {
      component = TestRenderer.create(<Router><YourLists showCreate={false}/></Router>);
    })

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Test Create List Button', () => {
  it('should render create list popout', async () => {
    let wrapper;
    await act( async () => {
      wrapper = mount(<Router><YourLists showCreate={false}/></Router>);
    })

    const createButton = wrapper.find('CreateButton');
    expect(createButton.length).toEqual(1);
    expect(createButton.prop('text')).toEqual("Create a New List");

    createButton.find('button').simulate('click');
    wrapper.update();
    expect(wrapper.find('form').length).toEqual(1);
  });
});
