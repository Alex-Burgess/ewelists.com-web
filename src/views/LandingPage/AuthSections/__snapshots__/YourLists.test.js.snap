// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for YourLists section. Renders your lists section 1`] = `
<div
  className="makeStyles-section-35"
>
  <div
    className="makeStyles-container-33"
  >
    <div
      className="MuiGrid-root makeStyles-grid-50  MuiGrid-container"
    >
      <div
        className="MuiGrid-root makeStyles-grid-154  MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12"
      >
        <h1
          className="makeStyles-title-36"
        >
          Your lists
        </h1>
        <br />
        <div
          className="MuiGrid-root makeStyles-grid-50  MuiGrid-container"
        >
          <div
            className="MuiGrid-root makeStyles-grid-154  MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4"
          >
            <div
              className="makeStyles-card-306 makeStyles-cardProfile-308"
            >
              <div
                className="makeStyles-cardHeader-321 makeStyles-cardHeaderImage-323"
              >
                <a
                  href="/lists/933c7476-cf0d-4a75-8822-2c011c07ce1d"
                >
                  <img
                    alt="..."
                    className="makeStyles-listImage-40"
                    src="/images/celebration-default.jpg"
                  />
                </a>
                <div
                  className="makeStyles-coloredShadow-1"
                  style={
                    Object {
                      "backgroundImage": "url(/images/celebration-default.jpg)",
                      "opacity": "1",
                    }
                  }
                />
              </div>
              <div
                className="makeStyles-cardBody-333"
              >
                <div
                  className="makeStyles-defaultFontStyle-340 makeStyles-infoText-347"
                >
                  <a
                    href="/lists/933c7476-cf0d-4a75-8822-2c011c07ce1d"
                  >
                    <h6
                      className="makeStyles-cardCategory-37"
                    >
                      My Birthday List
                    </h6>
                  </a>
                </div>
                <p
                  className="makeStyles-cardDescription-38"
                >
                  Birthday
                </p>
              </div>
              <div
                className="makeStyles-cardFooter-353 makeStyles-cardFooterProfile-354 makeStyles-justifyContentCenter-39"
              >
                <a
                  href="/lists/933c7476-cf0d-4a75-8822-2c011c07ce1d"
                >
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-159 makeStyles-info-163 makeStyles-round-186 makeStyles-justIcon-189"
                    disabled={false}
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M14 17H4v2h10v-2zm6-8H4v2h16V9zM4 15h16v-2H4v2zM4 5v2h16V5H4z"
                        />
                      </svg>
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </button>
                </a>
                <a
                  href="/edit/933c7476-cf0d-4a75-8822-2c011c07ce1d"
                >
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-159 makeStyles-success-164 makeStyles-round-186 makeStyles-justIcon-189"
                    disabled={false}
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      <span
                        aria-hidden={true}
                        className="material-icons MuiIcon-root"
                      >
                        mode_edit
                      </span>
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </button>
                </a>
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root makeStyles-grid-154  MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4"
          >
            <div
              className="makeStyles-addList-155"
            >
              <div>
                <div>
                  <div
                    className="makeStyles-centerButton-157"
                  >
                    <button
                      className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-159 makeStyles-lg-184 makeStyles-info-163 makeStyles-round-186 makeStyles-justIcon-189"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          className="MuiSvgIcon-root"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M14 10H2v2h12v-2zm0-4H2v2h12V6zm4 8v-4h-2v4h-4v2h4v4h2v-4h4v-2h-4zM2 16h8v-2H2v2z"
                          />
                        </svg>
                      </span>
                      <span
                        className="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <p
                    className="makeStyles-cardDescription-156"
                  >
                    Create a New List
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root makeStyles-grid-154  MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12"
      >
        <h1
          className="makeStyles-title-36"
        >
          Lists Shared With You
        </h1>
        <br />
        <div
          className="MuiGrid-root makeStyles-grid-50  MuiGrid-container"
        />
      </div>
    </div>
  </div>
</div>
`;
